src/plugins/odyssey/instance.js

```diff
@@ -0,0 +1,226 @@
+      return null;
+    }
+    const parts = NodeAddress.toParts(address);
+    // We know it is an OdysseyNodeTypeIdentifier because the instance's internal
+    // graph only has Odyssey nodes in it.
+    const nodeTypeIdentifier: OdysseyNodeTypeIdentifier = (parts[2]: any);
+    const description = NullUtil.get(this._descriptions.get(address));
+    return {address, nodeTypeIdentifier, description};
+  }
+
+  nodes(typeIdentifier?: OdysseyNodeTypeIdentifier): Iterator<Node> {
+    if (
+      typeIdentifier != null &&
+      !isOdysseyNodeTypeIdentifier(typeIdentifier)
+    ) {
+      throw new Error(`Invalid type identifier: ${typeIdentifier}`);
+    }
+    const prefix =
+      typeIdentifier == null
+        ? NODE_PREFIX
+        : NodeAddress.append(NODE_PREFIX, typeIdentifier);
+    return this._nodesIterator(prefix);
+  }
+
+  *_nodesIterator(prefix: NodeAddressT): Iterator<Node> {
+    for (const a of this._graph.nodes({prefix})) {
+      const node = NullUtil.get(this.node(a));
+      yield node;
+    }
+  }
+
+  // Cred for these edge flows from src to dst.
+  // So if e.g. a contribution was made by a person and supports a priority,
+  // there should be an edge from Priority -> Contribution and from
+  // Contribution -> Priority.
+  // TODO(@decentralion): Add support for edge types (also configured on a per-instance basis).
+  addEdge(
+    type: OdysseyEdgeTypeIdentifier,
+    src: Node,
+    dst: Node
+  ): OdysseyInstance {
+    if (!isOdysseyEdgeTypeIdentifier(type)) {
+      throw new Error(`Invalid Odyssey edge type identifier: ${type}`);
+    }
+    const edge = {
+      src: src.address,
+      dst: dst.address,
+      address: EdgeAddress.append(EDGE_PREFIX, type, String(this._count)),
+    };
+    this._graph.addEdge(edge);
+    this._count++;
+    return this;
+  }
+
+  /**
+   * Returns a copy of the graph underlying this instance.
+   */
+  graph(): Graph {
+    return this._graph.copy();
+  }
+
+  toJSON(): InstanceJSON {
+    const graphJSON = this._graph.toJSON();
+    const sortedNodeAddresses = sortedNodeAddressesFromJSON(graphJSON);
+    const sortedDescriptions = sortedNodeAddresses.map((a) =>
+      NullUtil.get(this._descriptions.get(a))
+    );
+    return toCompat(COMPAT_INFO, {
+      graphJSON,
+      sortedDescriptions,
+      count: this._count,
+    });
+  }
+
+  static fromJSON(j: InstanceJSON): OdysseyInstance {
+    const {graphJSON, sortedDescriptions, count} = fromCompat(COMPAT_INFO, j);
+    const instance = new OdysseyInstance();
+    instance._graph = Graph.fromJSON(graphJSON);
+    instance._count = count;
+    const descriptions = new Map();
+    const sortedNodeAddresses = sortedNodeAddressesFromJSON(graphJSON);
+    for (let i = 0; i < sortedNodeAddresses.length; i++) {
+      descriptions.set(sortedNodeAddresses[i], sortedDescriptions[i]);
+    }
+    instance._descriptions = descriptions;
+    return instance;
+  }
+
+  /**
+   * Returns whether two Odyssey instances are equal.
```

Comment:
consider renaming this from `equals` to `historicallyIdentical` because it's really ... not... equality